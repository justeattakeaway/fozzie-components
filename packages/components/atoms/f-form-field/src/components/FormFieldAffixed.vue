<template>
    <div
        :class="[
            $style['c-formField'],
            $style['c-formField--affixed'],
            $style[`c-formField--affixed--${fieldSize}`], {
                [$style['c-formField--invalid']]: hasError,
                [$style['c-formField--disabled']]: isDisabled
            }
        ]"
        :data-test-id="testId.affixed">
        <input
            :id="$attrs.id"
            ref="input"
            :class="$style['c-formField-field']"
            :disabled="attributes.disabled"
            v-bind="attributes"
            :value="value"
            :data-test-id="testId.input"
            @change="updateInput">

        <span
            v-if="prefix"
            :class="[
                $style['c-formField-affix'],
                $style['c-formField-prefix']
            ]"
            :data-test-id="testId.prefix"
            @click="addFocus">
            {{ prefix }}
        </span>

        <span
            v-if="suffix"
            :class="[
                $style['c-formField-affix'],
                $style['c-formField-suffix']
            ]"
            :data-test-id="testId.suffix"
            @click="addFocus">
            {{ suffix }}
        </span>
    </div>
</template>

<script>
export default {
    name: 'FormAffixedField',

    inheritAttrs: false,

    props: {
        attributes: {
            type: Object,
            default: () => {}
        },

        prefix: {
            type: String,
            default: ''
        },

        suffix: {
            type: String,
            default: ''
        },

        hasError: {
            type: Boolean,
            default: false
        },

        fieldSize: {
            type: String,
            default: 'medium'
        },

        value: {
            type: String,
            default: ''
        }
    },

    computed: {
        testId () {
            const formFieldName = (this.attributes && this.attributes.name ? this.attributes.name : null);

            return {
                affixed: formFieldName ? `formfield-${formFieldName}-affixed` : 'formfield-affixed',
                input: formFieldName ? `formfield-${formFieldName}-affixed-input` : 'formfield-affixed-input',
                prefix: formFieldName ? `formfield-${formFieldName}-affixed-prefix` : 'formfield-affixed-prefix',
                suffix: formFieldName ? `formfield-${formFieldName}-affixed-suffix` : 'formfield-affixed-suffix'
            };
        },

        isDisabled () {
            return this.attributes.disabled;
        }
    },

    methods: {
        addFocus () {
            this.$refs.input.focus();
        },

        updateInput (event) {
            this.$emit('update', event.target.value);
        }
    }
};
</script>

<style lang="scss" module>
$affixed-field-sizes: (
    small:  ($form-input-height--small, $form-input-padding--small),
    medium: ($form-input-height, $form-input-padding),
    large: ($form-input-height--large, $form-input-padding--large),
);

.c-formField--affixed {
    @extend %c-formField-field;
    display: flex;

    .c-formField-field {
        background-color: transparent;
        border: none;

        flex-grow: 1;
        order: 2;
        height: 100%;

        padding-left: 0;
        padding-right: 0;

        &:focus,
        &:active,
        &:focus-within {
            box-shadow: none;
        }

        &:hover,
        &:hover ~ .c-formField-affix {
            background-color: $form-input-bg--hover;
        }
    }
}

// Styling for either the prefix or suffix
.c-formField-affix {
    height: 100%;
    font-family: $font-family-base;
    @include font-size($form-input-fontSize);
    color: $form-input-secondaryTextColour;
}

.c-formField-prefix {
    order: 1;
}
.c-formField-suffix {
    order: 3;
}

@each $size, $properties in $affixed-field-sizes {
    .c-formField--affixed--#{$size} {
        height: nth($properties, 1);

        .c-formField-affix {
            padding: nth($properties, 2);
        }
    }
}

.c-formField--disabled {
    @include disabled-field();

    .c-formField-affix {
        @include disabled-field();
    }
}
</style>
