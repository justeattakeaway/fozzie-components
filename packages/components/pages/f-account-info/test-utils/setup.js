import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import { VueI18n } from '@justeat/f-globalisation';
import tenantConfigs from '../src/tenants';

export const localVue = createLocalVue();
localVue.use(VueI18n);
localVue.use(Vuex);

export const i18n = {
    locale: 'en-GB',
    messages: {
        'en-GB': tenantConfigs['en-GB'].messages
    }
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const consumerDetailsGetResponse = {
    data: {
        ConsumerId: 1050450174,
        Id: '8It2lEaJjG82nJtxO00Iab0CoF0=',
        CreatedDate: '2019-04-16T17:25:00.000Z',
        EmailAddress: 'satchin.moriyoka@example.com',
        FirstName: 'Satchin',
        LastName: 'Moriyoka',
        PhoneNumber: '+441234567890',
        DateOfBirth: '1914-04-16T17:25:00.000Z',
        TermsAndConditionsId: 0,
        WantsNewsletter: true,
        AccountType: 'Default',
        ConsumerStatus: 'ReturnConsumer'
    }
};

const consumerAddressGetResponse = {
    AddressId: 1050450174,
    City: 'Bristol',
    ZipCode: 'BS1 4DJ',
    AddressName: 'Office',
    IsDefault: true, // Key field
    Line1: 'Just Eat',
    Line2: 'Broad Quay House'
};

export const consumerAddressesGetResponse = {
    data: [{
        ...consumerAddressGetResponse
    },
    {
        AddressId: 1050450175,
        City: 'Bristol',
        ZipCode: 'BS1 9DJ',
        AddressName: 'Office',
        IsDefault: false,
        Line1: 'Takeaway',
        Line2: 'Broad Quay House'
    }]
};

export const consumerDetailsMappedModel = {
    id: consumerDetailsGetResponse.data.ConsumerId,
    firstName: consumerDetailsGetResponse.data.FirstName,
    lastName: consumerDetailsGetResponse.data.LastName,
    phoneNumber: consumerDetailsGetResponse.data.PhoneNumber,
    emailAddress: consumerDetailsGetResponse.data.EmailAddress
};

export const consumerAddressMappedModel = {
    line1: consumerAddressGetResponse.Line1,
    line2: consumerAddressGetResponse.Line2,
    line3: consumerAddressGetResponse.Line3,
    locality: consumerAddressGetResponse.City,
    postcode: consumerAddressGetResponse.ZipCode
};

export const consumerViewModel = {
    ...consumerDetailsMappedModel,
    ...consumerAddressMappedModel
};

export const consumerStateModel = {
    consumer: {
        ...consumerDetailsMappedModel,
        ...consumerAddressMappedModel
    }
};
