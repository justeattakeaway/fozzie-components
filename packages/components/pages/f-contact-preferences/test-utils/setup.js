export const contactPreferencesGetResponse = {
    data: {
        Preferences: [{
            DisplayName: 'Order status',
            Sort: 1,
            Key: 'orderstatus',
            Push: false,
            Email: true,
            Sms: true
        }, {
            DisplayName: 'Review meal',
            Sort: 2,
            Key: 'reviewmeal',
            Push: false,
            Email: false,
            Sms: false
        }, {
            DisplayName: 'News & offers',
            Sort: 3,
            Key: 'news',
            Push: false,
            Email: false,
            Sms: false
        }
        ],
        PreferenceVersionViewed: 0
    }
};

export const contactPreferencesViewModel = {
    preferenceVersionViewed: 0,
    preferences: [{
        displayName: 'Order status',
        isEmailEnabled: false,
        emailValue: true,
        key: 'orderstatus',
        pushEnabled: false,
        pushValue: false,
        isSmsEnabled: false,
        smsValue: true,
        sort: 1,
        visible: true
    }, {
        displayName: 'Review meal',
        isEmailEnabled: true,
        emailValue: false,
        key: 'reviewmeal',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 2,
        visible: false
    }, {
        displayName: 'News & offers',
        isEmailEnabled: true,
        emailValue: false,
        key: 'news',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 3,
        visible: true
    }]
};

export const filteredContactPreferencesModel = [{
    displayName: 'Order status',
    isEmailEnabled: false,
    emailValue: true,
    key: 'orderstatus',
    pushEnabled: false,
    pushValue: false,
    isSmsEnabled: false,
    smsValue: true,
    sort: 1,
    visible: true
}, {
    displayName: 'News & offers',
    isEmailEnabled: true,
    emailValue: false,
    key: 'news',
    pushEnabled: true,
    pushValue: false,
    isSmsEnabled: true,
    smsValue: false,
    sort: 3,
    visible: true
}];

export const contactPreferencesUpdateModel = {
    Preferences: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: false,
        Sms: false
    }],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const contactPreferencesUpdateBody = {
    Preference: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: true,
        Sms: false
    }
    ],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};
