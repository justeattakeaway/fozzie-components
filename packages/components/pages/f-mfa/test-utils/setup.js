export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

// eslint-disable-next-line camelcase
export const postBody = { mfa_token: 'some_mfa_token', otp: 'ABC123' };
