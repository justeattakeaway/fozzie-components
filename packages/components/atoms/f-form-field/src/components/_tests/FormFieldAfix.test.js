import { shallowMount } from '@vue/test-utils';
import FormFieldAffixed from '../FormFieldAffixed.vue';

describe('FormFieldAffixed', () => {
    const attributes = {
        disabled: null
    };

    const propsData = { attributes };

    it('should be defined', () => {
        const wrapper = shallowMount(FormFieldAffixed, { propsData });
        expect(wrapper.exists()).toBe(true);
    });

    describe('props ::', () => {
        describe('prefix ::', () => {
            describe.each([
                [null, false],
                ['', false],
                ['$', true]
            ])('when prefix value is', (prefixValue, prefixVisible) => {
                it(`should ${prefixVisible ? '' : 'not '}display 'c-formField-affixed-prefix'`, () => {
                    // Arrange & Act
                    propsData.prefix = prefixValue;

                    const wrapper = shallowMount(FormFieldAffixed, { propsData });
                    const formPrefix = wrapper.find('[data-test-id="formfield-affixed-prefix"]');

                    // Assert
                    expect(formPrefix.exists()).toBe(prefixVisible);
                });
            });
        });

        describe('suffix ::', () => {
            describe.each([
                [null, false],
                ['', false],
                ['GBP', true]
            ])('when suffix value is', (suffixValue, suffixVisible) => {
                it(`should ${suffixVisible ? '' : 'not '}display 'c-formField-affixed-suffix'`, () => {
                    // Arrange & Act
                    propsData.suffix = suffixValue;

                    const wrapper = shallowMount(FormFieldAffixed, { propsData });
                    const formSuffix = wrapper.find('[data-test-id="formfield-affixed-suffix"]');

                    // Assert
                    expect(formSuffix.exists()).toBe(suffixVisible);
                });
            });
        });
    });

    describe('computed ::', () => {
        describe('testId :: ', () => {
            describe('when `attributes` has name value', () => {
                it('should return testIds for form field', () => {
                    // Arrange
                    propsData.attributes = {
                        name: 'formfield-4'
                    };

                    // Act
                    const wrapper = shallowMount(FormFieldAffixed, { propsData });

                    // Assert
                    expect(wrapper.vm.testId).toMatchSnapshot();
                });
            });

            describe('when `attributes` does not have name value', () => {
                it('should return default testIds', () => {
                    // Arrange
                    propsData.attributes = {
                        name: null
                    };

                    // Act
                    const wrapper = shallowMount(FormFieldAffixed, { propsData });

                    // Assert
                    expect(wrapper.vm.testId).toMatchSnapshot();
                });
            });
        });
    });

    describe('methods ::', () => {
        describe('updateInput ::', () => {
            it('should emit `update` when an option is selected', async () => {
                // Arrange
                const wrapper = shallowMount(FormFieldAffixed, { propsData });

                const event = {
                    target: {
                        value: 'test option'
                    }
                };

                // Act
                await wrapper.vm.updateInput(event);

                // Assert
                expect(wrapper.emitted('update').length).toBe(1);
            });
        });
    });
});
