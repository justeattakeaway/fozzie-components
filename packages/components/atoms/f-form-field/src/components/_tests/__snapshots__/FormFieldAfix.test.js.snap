// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFieldAffixed computed :: testId ::  when \`attributes\` does not have name value should return default testIds 1`] = `
Object {
  "affixed": "formfield-affixed",
  "input": "formfield-affixed-input",
  "prefix": "formfield-affixed-prefix",
  "suffix": "formfield-affixed-suffix",
}
`;

exports[`FormFieldAffixed computed :: testId ::  when \`attributes\` has name value should return testIds for form field 1`] = `
Object {
  "affixed": "formfield-formfield-4-affixed",
  "input": "formfield-formfield-4-affixed-input",
  "prefix": "formfield-formfield-4-affixed-prefix",
  "suffix": "formfield-formfield-4-affixed-suffix",
}
`;
